cmake/test.cmake

```diff
@@ -157,19 +135,3 @@ function(add_unit_test NAME)
   add_cpp_test(${NAME} SRCS ${ARG_SRCS} LIBS ${ARG_LIBS} DEPS ${ARG_DEPS})
   add_dependencies(unit_tests "${NAME}_test")
 endfunction(add_unit_test)
-
-# Adds a component test named `${NAME}_test`, which is part of the `component_tests` target.
-#
-# The file `${NAME}_test.cc` is implicitly added as a source.
-# Additional source files can be declared with the `SRCS` parameter.
-#
-# Test executable is implicitly linked with `gtest` and `gmock`.
-# Additional libraries can be linked with the `LIBS` parameter.
-#
-# Additional dependencies can be declares with the `DEPS` parameter.
-add_custom_target(component_tests)
-function(add_component_test NAME)
-  cmake_parse_arguments(ARG "" "" "SRCS;LIBS;DEPS" ${ARGN})
-  add_cpp_test(${NAME} SRCS ${ARG_SRCS} LIBS ${ARG_LIBS} DEPS ${ARG_DEPS})
-  add_dependencies(component_tests "${NAME}_test")
-endfunction(add_component_test)
```

Comment:
This wasn't used and was adding an empty component_tests target which was a little confusing.  If we want to add different test categories in the future something similar can be added then.

Reply:
👍